### ISSUE TITLE
Adding xhtml:link alternates for multilingual website

### ISSUE THREAD

**Issue by nico0756** [harlan-zw/nuxt-simple-sitemap#20]
### Details

Hi,

I'm developing a multilingual website and I need to create a sitemap which contains the localized versions of every page. I found this issue [https://github.com/harlan-zw/nuxt-simple-sitemap/issues/11](url), so I tought it was possible and I created `server/api/_sitemap-urls.ts`:
```
export default cachedEventHandler(async e => {
    return [
        {
            loc: '/page-1'
        },
        {
            loc: '/page-2',
            links: [
                {
                    lang: 'en',
                    href: '/en/page-2'
                },
            ],
        }
    ];
}, {
    name: 'sitemap-dynamic-urls',
    maxAge: 60 * 10 // cache URLs for 10 minutes
})
```

But it generates this sitemap
```
<?xml version="1.0" encoding="UTF-8"?>
<?xml-stylesheet type="text/xsl" href="/__sitemap__/style.xsl"?>
<urlset xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:image="http://www.google.com/schemas/sitemap-image/1.1" xsi:schemaLocation="http://www.sitemaps.org/schemas/sitemap/0.9 http://www.sitemaps.org/schemas/sitemap/0.9/sitemap.xsd http://www.google.com/schemas/sitemap-image/1.1 http://www.google.com/schemas/sitemap-image/1.1/sitemap-image.xsd" xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
    <url>
        <loc>http://localhost:3000</loc>
    </url>
    <url>
        <loc>http://localhost:3000/page-1</loc>
    </url>
    <url>
        <loc>http://localhost:3000/page-2</loc>
        <links:links>
            <links:lang>en</links:lang>
            <links:href>/en/page-2</links:href>
        </links:links>
    </url>
</urlset>
<!-- XML Sitemap generated by Nuxt Simple Sitemap -->
```

Instead it should be something like
```
<url>
    <loc>http://localhost:3000/page-2</loc>
    <xhtml:link
               rel="alternate"
               hreflang="en"
               href="http://localhost:3000/en/page-2"/>
</url>
```

Am I doing something wrong or is it just not possible to generate a sitemap with xhtml:link elements?

**Comment by tmlmt** [comment]
Just bumped into this as I tried to implement it on my project and am discovering the details of sitemaps & localised versions. 
You're not doing anything wrong, it's just that Harlan's simple sitemap generator is only handling urls of type [SitemapFullEntry](https://github.com/harlan-zw/nuxt-simple-sitemap/blob/c5b582de6b8701e9a0d20c7cd69ad1cec3e51d4d/src/types.ts#L39) and doesn't handle the `xhtml` [namespace](https://www.w3.org/1999/xhtml/)

But if your goal is to indicate to search crawlers that there are localised versions of your pages, you can also do that by using HTML tags, as indicated in [^1]. And nuxt/i18n (if you're using it), [handles it for you](https://v8.i18n.nuxtjs.org/guide/seo). It's my understanding, when reading in [^1] where there is mention of three methods for doing so (HTML tags, HTTP headers and sitemap), that ...:
> you can use all three methods at the same time

... that this should be sufficient to cover both the generation of a proper sitemap and indicating localised versions. 
@harlan-zw could you confirm and give your thoughts on this issue?

[^1]: https://developers.google.com/search/docs/specialty/international/localized-versions

**Comment by harlan-zw** [comment]
Thanks for the issue @nico0756 and thanks for the detailed answer @tmlmt!

I intentionally didn't implement the `xhtml` namespace for two reasons:
- Wasn't sure if anyone would actually need it, using the meta tags / HTTP headers is generally easier
- I based the sitemap on the logic of [Yoast SEO](https://yoast.com/), which doesn't support this natively afaik

It seems like a valid feature so I'll look into supporting this soon.

**Comment by nico0756** [comment]
Thanks for the answers!

For now I'll stick to using the meta tags, but if in the future the xhtml namespace will be supported I'll try it for sure.

Thank you!

**Comment by heisig** [comment]
Thank you for putting this on the agenda. I would also like to see the xhtml namespace to be supported. It would be very useful to construct a sitemap with localized versions.

**Comment by harlan-zw** [comment]
Hey, this is now available in 2.4.0, you can use the `alternatives` key.

I've also added automatic @nuxtjs/i18n support as well which may be useful.

Let me know if you have any issues :)
